snapcraft/tests/test_lifecycle.py

```diff
@@ -300,3 +302,103 @@ def test_clean_part_already_clean(self, mock_exists, mock_rmtree):
             os.path.abspath(os.curdir), 'parts', part_name)
         mock_exists.assert_called_once_with(partdir)
         self.assertFalse(mock_rmtree.called)
+
+
+class CollisionTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        tmpdirObject = tempfile.TemporaryDirectory()
+        self.addCleanup(tmpdirObject.cleanup)
+        tmpdir = tmpdirObject.name
+
+        part1 = lifecycle.load_plugin('part1', 'nil', {})
+        part1.code.installdir = tmpdir + '/install1'
+        os.makedirs(part1.installdir + '/a')
+        open(part1.installdir + '/a/1', mode='w').close()
+
+        part2 = lifecycle.load_plugin('part2', 'nil', {})
+        part2.code.installdir = tmpdir + '/install2'
+        os.makedirs(part2.installdir + '/a')
+        with open(part2.installdir + '/1', mode='w') as f:
+            f.write('1')
+        open(part2.installdir + '/2', mode='w').close()
+        with open(part2.installdir + '/a/2', mode='w') as f:
+            f.write('a/2')
+
+        part3 = lifecycle.load_plugin('part3', 'nil', {})
+        part3.code.installdir = tmpdir + '/install3'
+        os.makedirs(part3.installdir + '/a')
+        os.makedirs(part3.installdir + '/b')
+        with open(part3.installdir + '/1', mode='w') as f:
+            f.write('2')
+        with open(part2.installdir + '/2', mode='w') as f:
+            f.write('1')
+        open(part3.installdir + '/a/2', mode='w').close()
+
+        self.part1 = part1
+        self.part2 = part2
+        self.part3 = part3
+
+    def test_no_collisions(self):
+        lifecycle._check_for_collisions([self.part1, self.part2])
```

Comment:
For tests like this, I like to put a comment: In case of error, _check_for_collisions will raise an exception.
Maybe to be extra pedantic, also docstring on _check_for_collisions the exception it raises, and when.